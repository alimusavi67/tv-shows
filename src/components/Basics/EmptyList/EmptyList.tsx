import React from 'react';

interface EmptyListProps {
  message: string;
}

const EmptyList: React.FC<EmptyListProps> = ({ message }) => {
  return (
    <div className="flex justify-center items-center h-full py-8">
      <div className="text-center text-gray-600 max-w-md mx-auto">
        <svg
          className="mx-auto h-12 w-12 text-gray-400"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="M9 20l-5.447-2.724A2 2 0 013 15.382V8.618a2 2 0 011.553-1.894L9 4m0 16l5.447 2.724A2 2 0 0015 21.382V14.618a2 2 0 00-1.553-1.894L9 10m0 16V10m0 0l5.447-2.724A2 2 0 0115 5.618V4.382a2 2 0 00-1.553-1.894L9 0"
          />
        </svg>
        <p className="mt-2">{message}</p>
      </div>
    </div>
  );
};

export default EmptyList;
