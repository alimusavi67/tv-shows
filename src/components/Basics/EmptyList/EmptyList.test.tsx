import React from 'react';
import { render } from '@testing-library/react';
import EmptyList from './EmptyList';

test('renders EmptyList correctly', () => {
  const { asFragment } = render(<EmptyList message="No items found" />);
  expect(asFragment()).toMatchSnapshot();
});
