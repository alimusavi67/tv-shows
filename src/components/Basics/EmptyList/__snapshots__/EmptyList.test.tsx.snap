// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EmptyList correctly 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center h-full py-8"
  >
    <div
      class="text-center text-gray-600 max-w-md mx-auto"
    >
      <svg
        class="mx-auto h-12 w-12 text-gray-400"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 20l-5.447-2.724A2 2 0 013 15.382V8.618a2 2 0 011.553-1.894L9 4m0 16l5.447 2.724A2 2 0 0015 21.382V14.618a2 2 0 00-1.553-1.894L9 10m0 16V10m0 0l5.447-2.724A2 2 0 0115 5.618V4.382a2 2 0 00-1.553-1.894L9 0"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <p
        class="mt-2"
      >
        No items found
      </p>
    </div>
  </div>
</DocumentFragment>
`;
